compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -966,7 +974,10 @@ fn report_unreachable_pattern<'p, 'tcx>(
             });
         }
         [covering_pat] if pat_is_catchall(covering_pat) => {
-            lint.covered_by_catchall = Some(covering_pat.data().span);
+            // A binding pattern that matches all, a single binding name.
```

Comment:
This comment is incorrect: `pat_is_catchall` also accepts patterns such as `(a, b)` or `Struct { field, .. }`. If you want only the case of a single binding/wildcard, check instead that `pat.ctor()` is `Constructor::Wildcard`.